Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -183,17 +183,27 @@ - (UIFont *)textMessageFont
     return [UIFont ows_dynamicTypeBodyFont];
 }
 
+- (UIFont *)tapForMoreFont
+{
+    return [UIFont ows_regularFontWithSize:12.f];
+}
+
+- (CGFloat)tapForMoreHeight
+{
+    return (CGFloat)ceil([self tapForMoreFont].lineHeight * 1.25);
+}
+
 - (OWSMessageCellType)cellType
 {
     return self.viewItem.messageCellType;
 }
 
-- (nullable NSString *)textMessage
+- (nullable DisplayableText *)displayableText
```

Comment:
I've introduced `DisplayableText` which has a text messages:

* Full, stripped and de-zalgo-ized text.
* Displayable (e.g. possibly truncated if very long) stripped de-zalgo-ized text.
* A flag indicating whether or not it was truncated (tap for more...).